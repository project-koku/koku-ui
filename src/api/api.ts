import { authInterceptor as insightsAuthInterceptor } from '@redhat-cloud-services/frontend-components-utilities/interceptors';
import axios, { AxiosRequestConfig } from 'axios';

export interface PagedMetaData {
  count: number;
}

export interface PagedLinks {
  first: string;
  previous: string;
  next: string;
  last: string;
}

export interface PagedResponse<D = any, M = any> {
  meta: M;
  links: PagedLinks;
  data: D[];
}

export function initApi({ version }: { version: string }) {
  axios.defaults.baseURL = `/api/cost-management/${version}/`;
  axios.interceptors.request.use(authInterceptor);
  axios.interceptors.request.use(insightsAuthInterceptor);
}

export function authInterceptor(reqConfig: AxiosRequestConfig): AxiosRequestConfig {
  return {
    ...reqConfig,
    headers: {
      ...reqConfig.headers,
    },
  };
}
