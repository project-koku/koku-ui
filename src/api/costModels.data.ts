export const page1 = {
  meta: {
    count: 20,
  },
  links: {
    first: '/api/cost-management/v1/cost-models/?limit=10&offset=0',
    next: null,
    previous: null,
    last: '/api/cost-management/v1/cost-models/?limit=10&offset=10',
  },
  data: [
    {
      uuid: '0f751bde-1b44-4031-8a9a-b704dad1f4a8',
      name: 'Cost Management AWS Cost Model',
      description: 'A cost model for markup on AWS costs.',
      source_type: 'Amazon Web Services',
      created_timestamp: '2020-12-06T20:09:33.706244Z',
      updated_timestamp: '2020-12-06T20:09:33.706266Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [
        {
          uuid: '9ae025a1-2f1b-48d0-a355-6859d143bdee',
          name: 'Test AWS Source',
        },
      ],
    },
    {
      uuid: '10058b62-feb8-4caf-956b-c5947cf58eba',
      name: 'Cost Management Azure Cost Model',
      description: 'A cost model for markup on Azure costs.',
      source_type: 'Microsoft Azure',
      created_timestamp: '2020-12-06T20:09:33.769610Z',
      updated_timestamp: '2020-12-06T20:09:33.769629Z',
      rates: [],
      markup: {
        value: '20.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [
        {
          uuid: 'd4f4ff82-efe7-4f31-b93c-6143282dec70',
          name: 'Test Azure Source',
        },
      ],
    },
    {
      uuid: 'bd6dac14-0cf1-469d-b478-251cf34a3207',
      name: 'Cost Management OpenShift Cost Model',
      description: 'A cost model of on-premises OpenShift clusters.',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.553720Z',
      updated_timestamp: '2020-12-07T11:33:12.066769Z',
      rates: [
        {
          metric: {
            name: 'cpu_core_usage_per_hour',
            label_metric: 'CPU',
            label_measurement: 'Usage',
            label_measurement_unit: 'core-hours',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 0.007,
            },
          ],
          cost_type: 'Supplementary',
        },
        {
          metric: {
            name: 'cpu_core_request_per_hour',
            label_metric: 'CPU',
            label_measurement: 'Request',
            label_measurement_unit: 'core-hours',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 0.2,
            },
          ],
          cost_type: 'Supplementary',
        },
        {
          metric: {
            name: 'memory_gb_usage_per_hour',
            label_metric: 'Memory',
            label_measurement: 'Usage',
            label_measurement_unit: 'GB-hours',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 0.009,
            },
          ],
          cost_type: 'Supplementary',
        },
        {
          metric: {
            name: 'memory_gb_request_per_hour',
            label_metric: 'Memory',
            label_measurement: 'Request',
            label_measurement_unit: 'GB-hours',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 0.05,
            },
          ],
          cost_type: 'Supplementary',
        },
        {
          metric: {
            name: 'storage_gb_usage_per_month',
            label_metric: 'Storage',
            label_measurement: 'Usage',
            label_measurement_unit: 'GB-month',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 0.01,
            },
          ],
          cost_type: 'Supplementary',
        },
        {
          metric: {
            name: 'storage_gb_request_per_month',
            label_metric: 'Storage',
            label_measurement: 'Request',
            label_measurement_unit: 'GB-month',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 0.01,
            },
          ],
          cost_type: 'Supplementary',
        },
        {
          metric: {
            name: 'node_cost_per_month',
            label_metric: 'Node',
            label_measurement: 'Currency',
            label_measurement_unit: 'node-month',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 1000,
            },
          ],
          cost_type: 'Infrastructure',
        },
        {
          metric: {
            name: 'cluster_cost_per_month',
            label_metric: 'Cluster',
            label_measurement: 'Currency',
            label_measurement_unit: 'cluster-month',
          },
          description: '',
          tiered_rates: [
            {
              unit: 'USD',
              usage: {
                unit: 'USD',
                usage_end: null,
                usage_start: null,
              },
              value: 10000,
            },
          ],
          cost_type: 'Infrastructure',
        },
        {
          metric: {
            name: 'pvc_cost_per_month',
            label_metric: 'Persistent volume claims',
            label_measurement: 'Currency',
            label_measurement_unit: 'pvc-month',
          },
          description: '',
          tag_rates: {
            tag_key: 'key-1',
            tag_values: [
              {
                unit: 'USD',
                value: 10,
                default: false,
                tag_value: 'value-2',
                description: '',
              },
            ],
          },
          cost_type: 'Infrastructure',
        },
      ],
      markup: {
        unit: 'percent',
      },
      source_uuids: [],
      sources: [
        {
          uuid: '774b88bd-6330-449d-a169-5a8bca7f9d35',
          name: 'Test OCP on Premises',
        },
      ],
    },
    {
      uuid: 'f3c81a95-be5a-4b40-82f3-6547609b95aa',
      name: 'Cost Management OpenShift on AWS Cost Model',
      description: 'A cost model for markup on OpenShift on AWS costs.',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.641427Z',
      updated_timestamp: '2020-12-06T20:09:33.641461Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [
        {
          uuid: 'e93a7ff6-49f5-45fc-aa92-3b4c358217cc',
          name: 'Test OCP on AWS',
        },
      ],
    },
    {
      uuid: 'd5c81a95-be5a-8b80-82f3-6547609b95aa',
      name: 'Cost Management OpenShift on AWS Cost Model 2',
      description: '',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.641427Z',
      updated_timestamp: '2020-12-06T20:09:33.641461Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [],
    },
    {
      uuid: 'eef81a95-be5a-4b40-82f3-9547609b95dd',
      name: 'Cost Management OpenShift on AWS Cost Model 3',
      description: '',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.641427Z',
      updated_timestamp: '2020-12-06T20:09:33.641461Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [],
    },
    {
      uuid: 'ffffff95-be5a-4b40-82f3-9547609b95dd',
      name: 'Cost Management OpenShift on AWS Cost Model 4',
      description: '',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.641427Z',
      updated_timestamp: '2020-12-06T20:09:33.641461Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [],
    },
    {
      uuid: '44444495-be5a-4b40-82f3-9547609b95dd',
      name: 'Cost Management OpenShift on AWS Cost Model 5',
      description: '',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.641427Z',
      updated_timestamp: '2020-12-06T20:09:33.641461Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [],
    },
    {
      uuid: '5aaaad95-ce5a-4b40-82f3-9547609b95dd',
      name: 'Cost Management OpenShift on AWS Cost Model 6',
      description: '',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.641427Z',
      updated_timestamp: '2020-12-06T20:09:33.641461Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [],
    },
    {
      uuid: '6dac7d9e-ce5a-4b40-82f3-9547609b95dd',
      name: 'Cost Management OpenShift on AWS Cost Model 7',
      description: '',
      source_type: 'OpenShift Container Platform',
      created_timestamp: '2020-12-06T20:09:33.641427Z',
      updated_timestamp: '2020-12-06T20:09:33.641461Z',
      rates: [],
      markup: {
        value: '10.0000000000',
        unit: 'percent',
      },
      source_uuids: [],
      sources: [],
    },
  ],
};

export const emptyPage = {
  meta: {
    count: 0,
  },
  links: {
    first: '/api/cost-management/v1/cost-models/?limit=10&offset=0',
    next: null,
    previous: null,
    last: '/api/cost-management/v1/cost-models/?limit=10&offset=0',
  },
  data: [],
};

export const noMatchPagePageNumber = {
  meta: {
    count: 0,
  },
  links: {
    first: '/api/cost-management/v1/cost-models/?limit=10&offset=20',
    next: null,
    previous: null,
    last: '/api/cost-management/v1/cost-models/?limit=10&offset=0',
  },
  data: [],
};

export const filterByAll = {
  meta: {
    count: 0,
  },
  links: {
    first:
      '/api/cost-management/v1/cost-models/?limit=10&offset=0&name=randomName&description=randomDesc&source_type=OCP',
    next: null,
    previous: null,
    last: '/api/cost-management/v1/cost-models/?limit=10&offset=0&name=randomName&description=randomDesc&source_type=OCP',
  },
  data: [],
};

export const noMatchPageName = {
  meta: {
    count: 0,
  },
  links: {
    first: '/api/cost-management/v1/cost-models/?limit=10&offset=0&name=randomName',
    next: null,
    previous: null,
    last: '/api/cost-management/v1/cost-models/?limit=10&offset=0&name=randomName',
  },
  data: [],
};
