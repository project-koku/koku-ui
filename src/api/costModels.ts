import axios from 'axios';

import { PagedResponse } from './api';
import { Rate, RateRequest } from './rates';

export interface CostModelProvider {
  name: string;
  uuid: string;
}

export interface CostModel {
  uuid: string;
  name: string;
  description: string;
  distribution: string;
  sources: CostModelProvider[];
  source_type: string;
  markup: { value: string; unit: string };
  rates: Rate[];
  created_timestamp: Date;
  updated_timestamp: Date;
}

export interface CostModelRequest {
  name: string;
  source_type: string;
  description: string;
  distribution: string;
  source_uuids: string[];
  rates: RateRequest[];
  markup: { value: string; unit: string };
}

export type CostModels = PagedResponse<CostModel>;

export function fetchCostModels(query = '') {
  return axios.get<CostModels>(`cost-models/${query && '?'}${query}`);
}

export function fetchCostModel(uuid: string) {
  return axios.get<CostModel>(`cost-models/${uuid}/`);
}

export function addCostModel(request: CostModelRequest) {
  return axios.post('cost-models/', request);
}

export function updateCostModel(uuid: string, request: CostModelRequest) {
  return axios.put(`cost-models/${uuid}/`, request);
}

export function deleteCostModel(uuid: string) {
  return axios.delete(`cost-models/${uuid}/`);
}
