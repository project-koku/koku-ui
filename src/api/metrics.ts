import { axiosInstance } from 'api';

import type { PagedResponse } from './api';

export interface Metric {
  name: string;
  source_type: string;
  metric: string;
  label_metric: string;
  label_measurement: string;
  label_measurement_unit: string;
  default_cost_type: string;
}

export interface MetricHash {
  [mtrc: string]: { [msrmnt: string]: Metric };
}

export type Metrics = PagedResponse<Metric>;

export function fetchRateMetrics(source_type = '') {
  const query = source_type ? `&source_type=${source_type}` : '';
  return axiosInstance.get<Metrics>(`metrics/?limit=1000${query}`);
}
