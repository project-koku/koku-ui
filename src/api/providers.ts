import axios from 'axios';

import { PagedMetaData, PagedResponse } from './api';

export interface ProviderAuthentication {
  uuid?: string;
  provider_resource_name: string;
}

export interface ProviderBillingSource {
  uuid?: string;
  bucket: string;
}

export interface ProviderCreatedBy {
  uuid?: string;
  username?: string;
  email?: string;
}

export interface ProviderCustomer {
  uuid?: string;
  name?: string;
  owner?: ProviderCreatedBy;
  date_created?: string;
}

export interface ProviderCostModel {
  name: string;
  uuid: string;
}

export interface Provider {
  active?: boolean;
  authentication?: ProviderAuthentication;
  billing_source?: ProviderBillingSource;
  created_by?: ProviderCreatedBy;
  created_timestamp?: Date;
  cost_models?: ProviderCostModel[];
  current_month_data?: boolean;
  customer?: ProviderCustomer;
  has_data?: boolean;
  name?: string;
  previous_month_data?: boolean;
  type?: string;
  uuid?: string;
}

export interface Providers extends PagedResponse<Provider, PagedMetaData> {}

// eslint-disable-next-line no-shadow
export const enum ProviderType {
  aws = 'aws',
  azure = 'azure',
  gcp = 'gcp',
  ibm = 'ibm',
  ocp = 'ocp',
}

export function fetchProviders(query: string) {
  const insights = (window as any).insights;
  const queryString = query ? `?${query}` : '';
  if (insights && insights.chrome && insights.chrome.auth && insights.chrome.auth.getUser) {
    return insights.chrome.auth.getUser().then(() => {
      return axios.get<Providers>(`sources/${queryString}`);
    });
  } else {
    return axios.get<Providers>(`sources/${queryString}`);
  }
}
