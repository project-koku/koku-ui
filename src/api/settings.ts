import { axiosInstance } from 'api';

import type { PagedLinks, PagedMetaData } from './api';

export interface SettingsData {
  clusters?: string[];
  default?: boolean;
  project?: string;
  group?: string;
  uuid?: string;
  key?: string;
  enabled?: boolean;
  source_type?: string;
}

export interface PagedMetaDataExt extends PagedMetaData {
  limit?: number;
  offset?: number;
  enabled_tags_count?: number;
  enabled_tags_limit?: number;
}

export interface Settings {
  meta: PagedMetaDataExt;
  links?: PagedLinks;
  data: SettingsData;
}

export interface SettingsPayload {
  parent?: string;
  children?: string[];
  ids?: string[];
}

// eslint-disable-next-line no-shadow
export const enum SettingsType {
  costCategories = 'costCategories',
  costCategoriesEnable = 'costCategoriesEnable',
  costCategoriesDisable = 'costCategoriesDisable',
  platformProjects = 'platformProjects',
  platformProjectsAdd = 'platformProjectsAdd',
  platformProjectsRemove = 'platformProjectsRemove',
  tags = 'tags',
  tagsEnable = 'tagsEnable',
  tagsDisable = 'tagsDisable',
  tagsMappings = 'tagsMappings',
  tagsMappingsChild = 'tagsMappingsChild',
  tagsMappingsChildAdd = 'tagsMappingsChildAdd',
  tagsMappingsChildRemove = 'tagsMappingsChildRemove',
  tagsMappingsParent = 'tagsMappingsParent',
  tagsMappingsParentRemove = 'tagsMappingsParentRemove',
}

export const SettingsTypePaths: Partial<Record<SettingsType, string>> = {
  [SettingsType.costCategories]: 'settings/aws_category_keys/',
  [SettingsType.costCategoriesEnable]: 'settings/aws_category_keys/enable/',
  [SettingsType.costCategoriesDisable]: 'settings/aws_category_keys/disable/',
  [SettingsType.platformProjects]: 'settings/cost-groups/',
  [SettingsType.platformProjectsAdd]: 'settings/cost-groups/add',
  [SettingsType.platformProjectsRemove]: 'settings/cost-groups/remove',
  [SettingsType.tags]: 'settings/tags',
  [SettingsType.tagsEnable]: 'settings/tags/enable/',
  [SettingsType.tagsDisable]: 'settings/tags/disable/',
  [SettingsType.tagsMappings]: 'settings/tags/mappings',
  [SettingsType.tagsMappingsChild]: 'settings/tags/mappings/child',
  [SettingsType.tagsMappingsChildAdd]: 'settings/tags/mappings/child/add',
  [SettingsType.tagsMappingsChildRemove]: 'settings/tags/mappings/child/remove',
  [SettingsType.tagsMappingsParent]: 'settings/tags/mappings/parent',
  [SettingsType.tagsMappingsParentRemove]: 'settings/tags/mappings/parent/remove',
};

export function fetchSettings(settingsType: SettingsType, query: string) {
  const path = SettingsTypePaths[settingsType];
  const queryString = query ? `?${query}` : '';
  return axiosInstance.get<Settings>(`${path}${queryString}`);
}

export function updateSettings(settingsType: SettingsType, payload: SettingsPayload) {
  const path = SettingsTypePaths[settingsType];
  return axiosInstance.put(`${path}`, payload);
}
