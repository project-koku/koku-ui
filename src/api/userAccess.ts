import axios from 'axios';

import { PagedLinks, PagedMetaData } from './api';

export interface UserAccessData {
  access?: boolean;
  type?: UserAccessType;
}

export interface UserAccess {
  meta: PagedMetaData;
  links?: PagedLinks;
  data: UserAccessData[] | boolean;
}

// eslint-disable-next-line no-shadow
export const enum UserAccessType {
  all = 'all',
  aws = 'aws',
  azure = 'azure',
  cost_model = 'cost_model',
  explorer = 'explorer',
  gcp = 'gcp',
  ibm = 'ibm',
  oci = 'oci',
  ocp = 'ocp',
}

// If the user-access API is called without a query parameter, all types are returned in the response
export function fetchUserAccess(query: string) {
  const insights = (window as any).insights;
  const queryString = query ? `?${query}` : '';
  if (insights && insights.chrome && insights.chrome.auth && insights.chrome.auth.getUser) {
    return insights.chrome.auth.getUser().then(() => {
      return axios.get<UserAccess>(`user-access/${queryString}`);
    });
  } else {
    return axios.get<UserAccess>(`user-access/${queryString}`);
  }
}
