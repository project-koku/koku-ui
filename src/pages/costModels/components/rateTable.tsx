import {
  compoundExpand,
  IActions,
  ISortBy,
  sortable,
  SortByDirection,
  Table,
  TableBody,
  TableHeader,
  TableVariant,
} from '@patternfly/react-table';
import { Rate } from 'api/rates';
import React from 'react';
import { formatCurrency } from 'utils/formatValue';

import { compareBy } from './rateForm/utils';
import TagRateTable from './tagRateTable';

interface RateTableProps {
  tiers: Rate[];
  actions?: IActions;
  isCompact?: boolean;
  t?: any;
}

export const RateTable: React.SFC<RateTableProps> = ({ t = v => v, tiers, actions, isCompact }) => {
  const [expanded, setExpanded] = React.useState({});
  const [sortBy, setSortBy] = React.useState<ISortBy>({});
  const cells = [
    { title: t('description') },
    { title: t('cost_models.table.metric'), transforms: [sortable] },
    { title: t('cost_models.table.measurement'), transforms: [sortable] },
    { title: t('cost_models.calculation_type') },
    { title: t('cost_models.rate'), cellTransforms: [compoundExpand] },
  ];
  const onSort = (_event, index: number, direction: SortByDirection) => {
    setSortBy({ index, direction });
  };
  let counter = 0;
  const rows = tiers
    .sort((r1, r2) => {
      const projection =
        sortBy.index === 1
          ? (r: Rate) => r.metric.label_metric
          : sortBy.index === 2
          ? (r: Rate) => r.metric.label_measurement
          : () => '';
      return compareBy(r1, r2, sortBy.direction, projection);
    })
    .reduce((acc, tier, ix) => {
      const rateKind = tier.tiered_rates ? 'regular' : 'tagging';
      let compoundRows = [];
      if (rateKind === 'tagging') {
        compoundRows = [
          {
            compoundParent: 4,
            parent: ix + counter,
            cells: [
              {
                title: <TagRateTable tagRates={tier.tag_rates} />,
                props: { colSpan: 6, className: 'pf-m-no-padding' },
              },
            ],
          },
        ];
        counter += 1;
      }
      const isOpen = rateKind === 'tagging' ? expanded[ix + counter - 1] || false : undefined;
      return [
        ...acc,
        {
          data: { index: ix, hasChildren: rateKind === 'tagging' },
          cells: [
            tier.description || '',
            tier.metric.label_metric,
            `${tier.metric.label_measurement} (${tier.metric.label_measurement_unit})`,
            tier.cost_type,
            {
              title:
                rateKind === 'regular'
                  ? `${formatCurrency(Number(tier.tiered_rates[0].value), 'USD')}`
                  : t('cost_models.table.tagged_rates'),
              props: { isOpen, style: { padding: rateKind === 'tagging' ? '' : '1.5rem 1rem' } },
            },
          ],
        },
        ...compoundRows,
      ];
    }, []);
  const onExpand = (_event: any, rowIndex: number, _colIndex: number, isOpen: boolean) => {
    setExpanded({ ...expanded, [rowIndex]: !isOpen });
  };
  const actionResolver = rowData => {
    if (rowData.compoundParent) {
      return [];
    }
    return actions;
  };
  return (
    <Table
      onSort={onSort}
      sortBy={sortBy}
      aria-label="price list"
      variant={isCompact ? TableVariant.compact : undefined}
      rows={rows}
      cells={cells}
      actionResolver={actionResolver}
      onExpand={onExpand}
    >
      <TableHeader />
      <TableBody />
    </Table>
  );
};
