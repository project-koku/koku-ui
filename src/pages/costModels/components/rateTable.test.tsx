import { fireEvent, render } from '@testing-library/react';
import { Rate } from 'api/rates';
import React from 'react';

import { RateTable } from './rateTable';

describe('rate-table', () => {
  // Todo: disabled until cost models is converted to react-intl
  xtest('smoke-test', () => {
    const tiers: Rate[] = [
      {
        description: 'rate 1',
        metric: {
          name: 'cpu_core_request_per_hour',
          metric: 'cpu_core_request_per_hour',
          source_type: 'openshift container platform',
          default_cost_type: 'Supplementary',
          label_metric: 'CPU',
          label_measurement: 'Request',
          label_measurement_unit: 'core-hour',
        },
        cost_type: 'Infrastructure',
        tiered_rates: [
          {
            unit: 'USD',
            value: 130.32,
            usage: {
              unit: 'USD',
            },
          },
        ],
      },
      {
        description: 'rate 2',
        metric: {
          name: 'cpu_core_request_per_hour',
          metric: 'cpu_core_request_per_hour',
          source_type: 'openshift container platform',
          label_metric: 'CPU',
          label_measurement: 'Usage',
          label_measurement_unit: 'core-hour',
          default_cost_type: 'Supplementary',
        },
        cost_type: 'Supplementary',
        tag_rates: {
          tag_key: 'openshift',
          tag_values: [
            {
              unit: 'USD',
              value: 0.43,
              tag_value: 'worker',
              description: 'default',
              default: true,
            },
            {
              unit: 'USD',
              value: 1.5,
              tag_value: 'grafana',
              description: 'grafana containers',
              default: false,
            },
          ],
        },
      },
    ];
    const { getByText } = render(<RateTable tiers={tiers} />);
    expect(getByText('rate 1')).toBeTruthy();
    expect(getByText('rate 2')).toBeTruthy();
    expect(getByText('grafana')).toBeTruthy();
  });
  test.skip('sort by metric & measurement', () => {
    // Todo: disabled until cost models is converted to react-intl
    // eslint-disable-next-line no-console
    console.error = jest.fn();
    const tiers: Rate[] = [
      {
        description: '',
        metric: {
          name: 'node_cost_per_month',
          metric: 'node_cost_per_month',
          source_type: 'openshift container platform',
          label_metric: 'Node',
          label_measurement: 'Currency',
          label_measurement_unit: 'node-month',
          default_cost_type: 'Supplementary',
        },
        cost_type: 'Supplementary',
        tiered_rates: [
          {
            unit: 'USD',
            value: 125.12,
            usage: {
              unit: 'USD',
            },
          },
        ],
      },
      {
        description: '',
        metric: {
          name: 'cpu_core_request_per_hour',
          metric: 'cpu_core_request_per_hour',
          source_type: 'openshift container platform',
          default_cost_type: 'Supplementary',
          label_metric: 'CPU',
          label_measurement: 'Request',
          label_measurement_unit: 'core-hour',
        },
        cost_type: 'Infrastructure',
        tiered_rates: [
          {
            unit: 'USD',
            value: 5.5,
            usage: {
              unit: 'USD',
            },
          },
        ],
      },
      {
        description: '',
        metric: {
          name: 'cpu_core_request_per_hour',
          metric: 'cpu_core_request_per_hour',
          source_type: 'openshift container platform',
          default_cost_type: 'Supplementary',
          label_metric: 'CPU',
          label_measurement: 'Usage',
          label_measurement_unit: 'core-hour',
        },
        cost_type: 'Infrastructure',
        tiered_rates: [
          {
            unit: 'USD',
            value: 7.2,
            usage: {
              unit: 'USD',
            },
          },
        ],
      },
      {
        description: '',
        metric: {
          name: 'cpu_core_request_per_hour',
          metric: 'cpu_core_request_per_hour',
          source_type: 'openshift container platform',
          default_cost_type: 'Supplementary',
          label_metric: 'CPU',
          label_measurement: 'Request',
          label_measurement_unit: 'core-hour',
        },
        cost_type: 'Supplementary',
        tiered_rates: [
          {
            unit: 'USD',
            value: 124.6,
            usage: {
              unit: 'USD',
            },
          },
        ],
      },
    ];
    const { queryAllByRole, getByRole } = render(<RateTable tiers={tiers} />);
    const metrics = queryAllByRole('cell', { name: /^(CPU|Node)$/ });
    expect(metrics).toMatchSnapshot();
    fireEvent.click(getByRole('button', { name: /metric/i }));
    expect(metrics).toMatchSnapshot();
    fireEvent.click(getByRole('button', { name: /metric/i }));
    expect(metrics).toMatchSnapshot();
    fireEvent.click(getByRole('button', { name: /measurement/i }));
    expect(metrics).toMatchSnapshot();
    fireEvent.click(getByRole('button', { name: /measurement/i }));
    expect(metrics).toMatchSnapshot();
  });
});
