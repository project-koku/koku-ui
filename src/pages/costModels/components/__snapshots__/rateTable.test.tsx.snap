// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rate-table sort by metric & measurement 1`] = `
Array [
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    Node
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
]
`;

exports[`rate-table sort by metric & measurement 2`] = `
Array [
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    Node
  </td>,
]
`;

exports[`rate-table sort by metric & measurement 3`] = `
Array [
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    Node
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
]
`;

exports[`rate-table sort by metric & measurement 4`] = `
Array [
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    Node
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
]
`;

exports[`rate-table sort by metric & measurement 5`] = `
Array [
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    CPU
  </td>,
  <td
    class=""
    data-key="1"
    data-label="EN Metric"
  >
    Node
  </td>,
]
`;
